Nostalgia/InRoomChat.cs

```diff
@@ -26,101 +27,148 @@ public void AddLine(string newLine)
 
     public void OnGUI()
     {
-        if (this.IsVisible && (PhotonNetwork.connectionStateDetailed == global::PeerState.Joined))
+        if (!this.IsVisible && (PhotonNetwork.connectionStateDetailed != global::PeerState.Joined))
         {
-            if ((Event.current.type == EventType.KeyDown) && ((Event.current.keyCode == KeyCode.KeypadEnter) || (Event.current.keyCode == KeyCode.Return)))
+            return;
+        }
+        if ((Event.current.type == EventType.KeyDown) && ((Event.current.keyCode == KeyCode.KeypadEnter) || (Event.current.keyCode == KeyCode.Return)))
+        {
+            if (!string.IsNullOrEmpty(this.inputLine))
             {
-                if (!string.IsNullOrEmpty(this.inputLine))
+                /**
+                 * This regex is used to parse input chat commands
+                 * @command and @args are used in "/command arg" type input
+                 * @single is used when "/command" type input
+                 * @tab because tab
+                 * @any when any normal chat input
+                 */
+                Match matches = Regex.Match(this.inputLine, @"\\|\/(?<command>\w*)\s(?<args>[^\s].*)|\\|\/(?<single>\w*)|(?<tab>\t)|(?<any>^[^\\\/].*?$)");
+
+                if (matches.Groups["tab"].Value != "")
                 {
-                    if (this.inputLine == "\t")
-                    {
-                        this.inputLine = string.Empty;
-                        GUI.FocusControl(string.Empty);
-                        return;
-                    }
-                    if ((this.inputLine == "/restart") && PhotonNetwork.isMasterClient)
-                    {
-                        this.inputLine = string.Empty;
-                        GUI.FocusControl(string.Empty);
-                        GameObject.Find("MultiplayerManager").GetComponent<FengGameManagerMKII>().restartGame(false);
-                        return;
-                    }
-                    if ((this.inputLine.Length <= 7) || !(this.inputLine.Substring(0, 7) == "/kick #"))
-                    {
-                        object[] parameters = new object[] { this.inputLine, LoginFengKAI.player.name };
-                        GameObject.Find("MultiplayerManager").GetComponent<FengGameManagerMKII>().photonView.RPC("Chat", PhotonTargets.All, parameters);
-                    }
-                    else if (this.inputLine.Remove(0, 7) == PhotonNetwork.masterClient.ID.ToString())
-                    {
-                        GameObject.Find("Chatroom").GetComponent<InRoomChat>().addLINE("error:can't kick master client.");
-                    }
-                    else if (this.inputLine.Remove(0, 7) == PhotonNetwork.player.ID.ToString())
+                    this.inputLine = string.Empty;
+                    GUI.FocusControl(string.Empty);
+                    return;
+                }
+
+                // Add normal commands without arguments
+                if (matches.Groups["single"].Value != "")
```

Comment:
I'd prefer if you were to replace the use of the tab with a:
- If chat is empty or has only spaces and you press enter; nothing happens and the chat closes.
- If chat has something written and you press enter: you check for command and either execute or send message